src/rbr/RBR.java

```diff
@@ -134,22 +134,28 @@ private void adjustRegions(Vertex sw) {
         ArrayList<Region> newRegions = new ArrayList<>();
         ArrayList<Region> regionsToBeRemoved = new ArrayList<>();
         for (Region currentRegion : regionsForVertex.get(sw)) {
-            Set<Vertex> outsiders = currentRegion.outsiders();
-            if(outsiders.isEmpty())
-                continue;
-            Range outsidersBox = TopologyKnowledge.box(outsiders);
-            Set<Vertex> trulyDestinationsInOutsidersRange = currentRegion.destinationsIn(outsidersBox);
-
+            newRegions.addAll(adjustRegions(currentRegion));
             regionsToBeRemoved.add(currentRegion);
-            ArrayList<Region> regionsToAdd = splitRegionExcludingOutsiders(currentRegion, outsidersBox);
-            newRegions.addAll(regionsToAdd);
-            // use others routers to make others regions
-            newRegions.addAll(makeRegions(trulyDestinationsInOutsidersRange, currentRegion.inputPorts(), currentRegion.outputPorts()));
         }
         regionsForVertex.get(sw).removeAll(regionsToBeRemoved);
         regionsForVertex.get(sw).addAll(newRegions);
     }
 
+    private ArrayList<Region> adjustRegions(Region r) {
+        ArrayList<Region> newRegions = new ArrayList<>();
+        Set<Vertex> outsiders = r.outsiders();
+        assert (!outsiders.isEmpty()) : "Outsiders is Empty";
```

Comment:
You shouldn't assert that. 'outsiders' can be empty sometimes and it means that you don't need to adjust that region 'r'. You could in that case just add 'r' to 'newRegions' and return it.